examples/autoignition-mpi.py

```diff
@@ -0,0 +1,244 @@
+    #    print(f"Initial CV rhoY: {cv.mass_fractions}")
+    #    print(f"Initial Y: {cv.mass_fractions / cv.mass}")
+    #    print(f"Initial DV pressure: {eos.pressure(cv)}")
+    #    print(f"Initial DV temperature: {eos.temperature(cv)}")
+
+    def my_chem_sources(discr, q, eos):
+        cv = split_conserved(dim, q)
+        omega = eos.get_production_rates(cv)
+        w = eos.get_species_molecular_weights()
+        species_sources = w * omega
+        rho_source = 0 * cv.mass
+        mom_source = 0 * cv.momentum
+        energy_source = 0 * cv.energy
+        return join_conserved(dim, rho_source, energy_source, mom_source,
+                              species_sources)
+    sources = {my_chem_sources}
+
+    visualizer = make_visualizer(discr, discr.order + 3
+                                 if discr.dim == 2 else discr.order)
+    initname = initializer.__class__.__name__
+    eosname = eos.__class__.__name__
+    init_message = make_init_message(dim=dim, order=order,
+                                     nelements=local_nelements,
+                                     global_nelements=global_nelements,
+                                     dt=current_dt, t_final=t_final, nstatus=nstatus,
+                                     nviz=nviz, cfl=current_cfl,
+                                     constant_cfl=constant_cfl, initname=initname,
+                                     eosname=eosname, casename=casename)
+    cantera_soln.equilibrate("UV")
+    can_eq_t, can_eq_rho, can_eq_y = cantera_soln.TDY
+    can_eq_p = cantera_soln.P
+    #    can_eq_e = cantera_soln.int_energy_mass
+    #    can_eq_k = cantera_soln.forward_rate_constants
+    #    can_eq_c = cantera_soln.concentrations
+    #    can_eq_r = cantera_soln.net_rates_of_progress
+    #    can_eq_omega = cantera_soln.net_production_rates
+    if rank == 0:
+        logger.info(init_message)
+        logger.info(f"Expected (p,T,rho, y) = ({can_eq_p}, {can_eq_t},"
+                    f" {can_eq_rho}, {can_eq_y})")
+
+    get_timestep = partial(inviscid_sim_timestep, discr=discr, t=current_t,
+                           dt=current_dt, cfl=current_cfl, eos=eos,
+                           t_final=t_final, constant_cfl=constant_cfl)
+
+    def my_rhs(t, state):
+        return inviscid_operator(discr, q=state, t=t,
+                                 boundaries=boundaries, eos=eos,
+                                 sources=sources)
+
+    def my_checkpoint(step, t, dt, state):
+        global checkpoint_t
+        cv = split_conserved(dim, state)
+        reaction_rates = eos.get_production_rates(cv)
+        viz_fields = [("reaction_rates", reaction_rates)]
+        return sim_checkpoint(discr, visualizer, eos, q=state,
+                              vizname=casename, step=step,
+                              t=t, dt=dt, nstatus=nstatus, nviz=nviz,
+                              constant_cfl=constant_cfl, comm=comm,
+                              viz_fields=viz_fields)
```

Comment:
This interface allows users to customize the fields that go into the viz files.  In this case stuffing `reaction_rates` into `viz_fields` does the trick.